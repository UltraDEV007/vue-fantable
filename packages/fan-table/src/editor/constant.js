// instance methods
export const INSTANCE_METHODS = {
  TEXTAREA_ADD_NEW_LINE: 'textareaAddNewLine',
  TEXTAREA_SELECT: 'textareaSelect',
}
