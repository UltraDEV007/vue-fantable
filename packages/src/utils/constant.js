// key codes
export const KEY_CODES = {
  BACK_SPACE: 8,
  TAB: 9,
  ENTER: 13,
  SHIFT: 16,
  SPACE: 32,

  ARROW_LEFT: 37,
  ARROW_UP: 38,
  ARROW_RIGHT: 39,
  ARROW_DOWN: 40,

  DELETE: 46,

  F2: 113,
}

// icon name
export const ICON_NAMES = {
  FILTER: 'filter',

  DOUBLE_RIGHT_ARROW: 'double-right-arrow',
  DOUBLE_LEFT_ARROW: 'double-left-arrow',

  TOP_ARROW: 'top-arrow',
  RIGHT_ARROW: 'right-arrow',
  BOTTOM_ARROW: 'bottom-arrow',
  LEFT_ARROW: 'left-arrow',

  SORT_TOP_ARROW: 'sort-top-arrow',
  SORT_BOTTOM_ARROW: 'sort-bottom-arrow',

  SEARCH: 'search',
}

// mouse event key type
export const MOUSE_EVENT_CLICK_TYPE = {
  LEFT_MOUSE: 1,
  MIDDLE_MOUSE: 2,
  RIGHT_MOUSE: 2,
}
