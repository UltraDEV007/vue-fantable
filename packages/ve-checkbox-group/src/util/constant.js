// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
