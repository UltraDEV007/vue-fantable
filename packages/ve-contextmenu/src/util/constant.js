// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
