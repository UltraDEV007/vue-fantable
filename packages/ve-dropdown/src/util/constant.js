// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
