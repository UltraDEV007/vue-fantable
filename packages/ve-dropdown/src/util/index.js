/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
