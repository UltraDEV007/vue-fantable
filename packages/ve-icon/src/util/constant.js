// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
