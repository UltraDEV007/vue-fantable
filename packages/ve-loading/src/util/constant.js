// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
