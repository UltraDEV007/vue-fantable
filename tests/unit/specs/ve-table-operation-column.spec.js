import { mount } from "@vue/test-utils";
import veTable from "@/ve-table";
import { later } from "../util";
import { KEY_CODES } from "../constant";

describe("veTable operation column", () => {
    const TABLE_DATA = [
        {
            name: "John",
            date: "1900-05-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Shanghai",
            rowKey: "1",
        },
        {
            name: "Dickerson",
            date: "1910-06-20",
            hobby: "coding",
            address: "No.1 Century Avenue, Beijing",
            rowKey: "2",
        },
        {
            name: "Larsen",
            date: "2000-07-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Chongqing",
            rowKey: "3",
        },
        {
            name: "Geneva",
            date: "2010-08-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Xiamen",
            rowKey: "4",
        },
        {
            name: "Jami",
            date: "2020-09-20",
            hobby: "coding and coding repeat",
            address: "No.1 Century Avenue, Shenzhen",
            rowKey: "5",
        },
    ];

    const COLUMNS = [
        {
            field: "index",
            key: "index",
            operationColumn: true,
            title: "#",
            width: 50,
            align: "center",
            renderBodyCell: ({ row, column, rowIndex }, h) => {
                return ++rowIndex;
            },
            edit: true,
        },
        {
            field: "name",
            key: "a",
            title: "Name",
            align: "left",
            width: "20%",
        },
        {
            field: "date",
            key: "b",
            title: "Date",
            align: "left",
            width: "20%",
        },
        {
            field: "hobby",
            key: "c",
            title: "Hobby",
            align: "center",
            width: "30%",
        },
        { field: "address", key: "d", title: "Address", width: "30%" },
    ];

    it("render", () => {
        const WRAPPER = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                cellSelectionOption: {
                    // default true
                    enable: true,
                },
                rowKeyFieldName: "rowKey",
            },
        });
        expect(WRAPPER.html()).toMatchSnapshot();
    });

    it("operation column", async () => {
        const wrapper = mount(veTable, {
            propsData: {
                columns: COLUMNS,
                tableData: TABLE_DATA,
                rowKeyFieldName: "rowKey",
            },
        });

        await later();

        const selectionTd = wrapper
            .findAll(".ve-table-body-tr")
            .at(0)
            .findAll(".ve-table-body-td")
            .at(0);

        expect(selectionTd.classes()).toContain("ve-table-operation-col");
    });
});
