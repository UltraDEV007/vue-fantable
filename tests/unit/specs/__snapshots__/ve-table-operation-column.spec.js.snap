// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`veTable operation column > render 1`] = `
"<div class="vue-table-root">
  <div style="width: 100%;" class="fan-table fan-table-border-around">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <!---->
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 50px;">
            <col style="width: 20%;">
            <col style="width: 20%;">
            <col style="width: 30%;">
            <col style="width: 30%;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header" tableviewportwidth="0">
            <tr class="fan-table-header-tr">
              <th style="text-align: center; left: 0px; top: 0px;" class="fan-table-header-th fan-table-fixed-left fan-table-last-left-fixed-column" rowspan="1" colspan="1">
                <!---->#
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: left; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Name
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: left; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Date
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: center; top: 0px;" class="fan-table-header-th" rowspan="1" colspan="1">
                <!---->Hobby
                <!---->
                <!---->
                <!---->
              </th>
              <th style="text-align: center; top: 0px;" class="fan-table-header-th fan-table-last-column" rowspan="1" colspan="1">
                <!---->Address
                <!---->
                <!---->
                <!---->
              </th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-stripe fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr class="fan-table-body-tr" row-key="1">
              <td class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;" rowspan="1" colspan="1" col-key="index">
                <!---->
                <!---->
                <!---->1
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->John
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->1900-05-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Shanghai
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="2">
              <td class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;" rowspan="1" colspan="1" col-key="index">
                <!---->
                <!---->
                <!---->2
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Dickerson
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->1910-06-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Beijing
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="3">
              <td class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;" rowspan="1" colspan="1" col-key="index">
                <!---->
                <!---->
                <!---->3
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Larsen
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->2000-07-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Chongqing
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="4">
              <td class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;" rowspan="1" colspan="1" col-key="index">
                <!---->
                <!---->
                <!---->4
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Geneva
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->2010-08-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Xiamen
              </td>
            </tr>
            <!---->
            <tr class="fan-table-body-tr" row-key="5">
              <td class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;" rowspan="1" colspan="1" col-key="index">
                <!---->
                <!---->
                <!---->5
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="a">
                <!---->
                <!---->
                <!---->Jami
              </td>
              <td class="fan-table-body-td" style="text-align: left;" rowspan="1" colspan="1" col-key="b">
                <!---->
                <!---->
                <!---->2020-09-20
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="c">
                <!---->
                <!---->
                <!---->coding and coding repeat
              </td>
              <td class="fan-table-body-td" style="text-align: center;" rowspan="1" colspan="1" col-key="d">
                <!---->
                <!---->
                <!---->No.1 Century Avenue, Shenzhen
              </td>
            </tr>
            <!---->
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="fan-table-selection-middle">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <div class="fan-table-selection-fixed-right">
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
        </div>
      </div>
    </div>
    <div style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;" class="fan-table-edit-input-container" allrowkeys="1,2,3,4,5"><textarea class="fan-table-edit-input" tabindex="-1"></textarea></div>
    <!---->
    <!---->
  </div>
</div>"
`;

exports[`veTable operation column render 1`] = `
<div class="vue-table-root">
  <div class="fan-table fan-table-border-around" style="width: 100%;">
    <div class="fan-table-container fan-table-enable-cell-selection" style="height: 0px;">
      <div class="fan-table-content-wrapper">
        <table class="fan-table-content fan-table-border-x">
          <colgroup>
            <col style="width: 50px;">
            <col style="width: 20%;">
            <col style="width: 20%;">
            <col style="width: 30%;">
            <col style="width: 30%;">
          </colgroup>
          <thead class="fan-table-fixed-header fan-table-header">
            <tr class="fan-table-header-tr">
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-fixed-left fan-table-last-left-fixed-column" style="text-align: center; left: 0px; top: 0px;">#</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Name</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: left; top: 0px;">Date</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th" style="text-align: center; top: 0px;">Hobby</th>
              <th rowspan="1" colspan="1" class="fan-table-header-th fan-table-last-column" style="text-align: center; top: 0px;">Address</th>
            </tr>
          </thead>
          <tbody class="fan-table-body fan-table-row-hover fan-table-row-highlight">
            <tr style="height: 0px;">
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
              <td style="padding: 0px; border: 0px; height: 0px;"></td>
            </tr>
            <tr row-key="1" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="index" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">1</td>
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">John</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1900-05-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">No.1 Century Avenue, Shanghai</td>
            </tr>
            <tr row-key="2" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="index" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">2</td>
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Dickerson</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">1910-06-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">No.1 Century Avenue, Beijing</td>
            </tr>
            <tr row-key="3" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="index" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">3</td>
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Larsen</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2000-07-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">No.1 Century Avenue, Chongqing</td>
            </tr>
            <tr row-key="4" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="index" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">4</td>
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Geneva</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2010-08-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">No.1 Century Avenue, Xiamen</td>
            </tr>
            <tr row-key="5" class="fan-table-body-tr">
              <td rowspan="1" colspan="1" col-key="index" class="fan-table-body-td fan-table-fixed-left fan-table-operation-col fan-table-last-left-fixed-column" style="text-align: center; left: 0px;">5</td>
              <td rowspan="1" colspan="1" col-key="a" class="fan-table-body-td" style="text-align: left;">Jami</td>
              <td rowspan="1" colspan="1" col-key="b" class="fan-table-body-td" style="text-align: left;">2020-09-20</td>
              <td rowspan="1" colspan="1" col-key="c" class="fan-table-body-td" style="text-align: center;">coding and coding repeat</td>
              <td rowspan="1" colspan="1" col-key="d" class="fan-table-body-td" style="text-align: center;">No.1 Century Avenue, Shenzhen</td>
            </tr>
          </tbody>
          <tfoot class="fan-table-fixed-footer fan-table-footer"></tfoot>
        </table>
        <div class="fan-table-selection-wrapper">
          <div class="fan-table-selection-fixed-left"></div>
          <div class="fan-table-selection-middle"></div>
          <div class="fan-table-selection-fixed-right"></div>
        </div>
      </div>
    </div>
    <div class="fan-table-edit-input-container" style="top: 0px; left: 0px; height: 1px; z-index: -1; opacity: 0;"><textarea tabindex="-1" class="fan-table-edit-input"></textarea></div>
  </div>
</div>
`;
